gym/envs/box2d/lunar_lander.py

```diff
@@ -489,11 +490,13 @@ def step(self, action):
         dispersion = [self.np_random.uniform(-1.0, +1.0) / SCALE for _ in range(2)]
 
         m_power = 0.0
-        if (self.continuous and action[0] > 0.0) or (
+
+        if (self.continuous and action[0] > 0.0) or (  # type: ignore
             not self.continuous and action == 2
         ):
             # Main engine
             if self.continuous:
+                assert isinstance(action, np.ndarray)
```

Comment:
Im unsure about this, as pytorch or tensorflow Tensors would fail this

Reply:
If a tensor is passed to the `step` method, the `np.clip` operation done previously ensures that it's converted to a numpy array. 
Moreover, the `step` method should only accept numpy arrays for the continuous case and an `int` for the discrete case, and users should ensure to supply the correct types (converting their tensors into numpy arrays).

Reply:
Where is this `np.clip` happening? I can't see it in `step`

Reply:
Sorry, I was talking about the car racing not lunar lander I think this is good in lunar lander then